Use vendored `fluxc` and `login`

wzieba opened the pull request:
### Description 

This PR introduces necessary changes to make the `assembleWasabiDebug` pass when using vendored `fluxc` and `login` modules.

Among others, this PR:
  - replaces coordinates of `fluxc` and `login` to point to local modules
  - updates dependencies and adds missing ones for vendored modules
  - unifies JVM version
  - adds necessary dependencies
  - fixes minor code issues, forced by Kotlin update on vendored codebase

The current state of PR is based on code from following commits
- FluxC: 2.103.0 [3061745](https://github.com/wordpress-mobile/WordPress-FluxC-Android/commit/3061745a6df23e3f137b088b929a7dfe6450a0f4)
- Login: 1.21.0 [72b7552](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/72b75520324a3ae7c132a97ae238c7db83bc62f6)

### Extraction from FluxC and Login

To perform extractions/filtering, I've used the following script. 

> [!CAUTION]
>  **DO NOT RUN** the script, it's only for preview and understanding what actions has been performed via `git filter-repo`

<details><summary>script</summary>

```shell

#!/bin/bash

# Prepare FluxC

cd ~

rm -rf WordPress-FluxC-Android

git clone git@github.com:wordpress-mobile/WordPress-FluxC-Android.git

cd WordPress-FluxC-Android

git checkout 3061745

git filter-repo \
--path-rename example/build.gradle:fluxc-tests/build.gradle \
--path-rename example/src/test:fluxc-tests/src/test \
--force

git filter-repo \
--path fluxc-annotations \
--path fluxc-processor \
--path fluxc \
--path plugins/woocommerce \
--path fluxc-tests \
--force

git filter-repo \
--path-rename fluxc-annotations:libs/fluxc-annotations \
--path-rename fluxc-processor:libs/fluxc-processor \
--path-rename plugins/woocommerce:libs/fluxc-plugin \
--path-rename fluxc:libs/fluxc \
--path-rename fluxc-tests:libs/fluxc-tests

git tag | xargs git tag -d

git checkout -b filtered

cd ~/woocommerce-android

git checkout trunk

git pull

git branch -D fluxc_with_history

git checkout -b fluxc_with_history

git pull fluxcfiltered filtered --allow-unrelated-histories --no-rebase

git push --set-upstream origin fluxc_with_history -f

## Prepare Login

cd ~

rm -rf WordPress-Login-Flow-Android

git clone git@github.com:wordpress-mobile/WordPress-Login-Flow-Android.git

cd WordPress-Login-Flow-Android

git checkout 72b7552 

git filter-repo --path WordPressLoginFlow --force

git filter-repo --path-rename WordPressLoginFlow:libs/login --force

git tag | xargs git tag -d

git checkout -b filtered

cd ~/woocommerce-android

git checkout fluxc_with_history

git branch -D login_with_history

git checkout -b login_with_history

git pull loginflowfilter filtered --allow-unrelated-histories --no-rebase

git push --set-upstream origin login_with_history -f

## Update adjustment branch

git checkout login_with_history

git branch -D adjust_fluxc_login_vendor

git checkout -b adjust_fluxc_login_vendor

# Now cherry pick adjustments commits manually

```

</details> 

<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

### Testing information

The whole app needs to be tested: nothing in particular.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

wzieba committed:
Use vendored `fluxc` and `login`

This PR introduces necessary changes to make the `assembleWasabiDebug` pass when using vendored `fluxc` and `login` modules.

Among others, this PR:
  - replaces coordinates of `fluxc` and `login` to point to local modules
  - updates dependencies and adds missing ones for vendored modules
  - unifies JVM version
  - adds necessary dependencies
  - fixes minor code issues, forced by Kotlin update on vendored codebase

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

wzieba committed:
Introduce `fluxc-tests` module

Which contains all unit tests from original `example` module. Also, this commit adds a number of changes to make unit tests run.

`ArrayUtilsTest` is removed as `ArrayUtils` was an example-app only class.

wzieba committed:
Upgrade `mockito`, remove `mockito-inline`

- `mockito` upgrade is necessary to make the build work with newer JDKs

```
Java 21 (65) is not supported by the current version of Byte Buddy which officially supports Java 20 (64) - update Byte Buddy or set net.bytebuddy.experimental as a VM property
```

- since 5.0, `mockito-core` includes the `mockito-inline` module

wzieba committed:
Mock a response from `wooNetwork`

To address `NoWhenBranchMatchedException`: these failing tests were crashing because `response` in `ProductRestClient#fetchProducts` was checked in `when` branch, while the value was null (not mocked). This change was requested by Kotlin update.

wzieba committed:
Fix invalid method mapping in compileSdk 35

Citing the documentation:

```
The List type in Java is mapped to the MutableList type in Kotlin. Because the List.removeFirst() and List.removeLast() APIs have been introduced in Android 15 (API level 35), the Kotlin compiler resolves function calls, for example list.removeFirst(), statically to the new List APIs instead of to the extension functions in kotlin-stdlib.

If an app is re-compiled with compileSdk set to 35 and minSdk set to 34 or lower, and then the app is run on Android 14 and lower, a runtime error is thrown:
```

https://developer.android.com/about/versions/15/behavior-changes-15

wzieba committed:
Disable `formatting` rules from fluxc and login modules

They were not enabled/configured back in FluxC, so enabling them here causes hundreds of violations.

Adjust `MagicNumber` rule to also fit newly added codebase

wzieba committed:
Remove unused commented code from `fluxc-tests`

wzieba committed:
Disable `allWarningsAsErrors` only in `fluxc`, `fluxc-plugin`, `fluxc-tests` and `login` modules

This option was not enabled originally.

wzieba commented:
@ParaskP7 this PR is ready for the first round of reviews. The chain of branches is following:

`trunk` -> `fluxc_with_history` -> `login_with_history` -> `adjust_fluxc_login_vendor`

`*_history` branches are result of `git filter-repo` - I'll prepare PRs for them later, but they don't have to be really reviewed as it was with WPAndroid.

----

I'm working right now on running benchmarks to make sure that we don't experience any performance regression with incremental builds. After this, and syncing with Woo Android team, this work will be ready to be merged.

ParaskP7 commented on config/detekt/detekt.yml:
```diff
@@ -15,6 +15,7 @@ complexity:
 formatting:
   active: true
   autoCorrect: true
+  excludes: ['**/fluxc/**', '**/login/**']
```
Praise (❤️): Nice! 👍

ParaskP7 commented on config/detekt/detekt.yml:
```diff
@@ -27,6 +28,8 @@ style:
     ignoreAnnotated: [ 'Module' ]
   MagicNumber:
+    ignoreEnums: true
+    ignorePropertyDeclaration: true
```
Praise (❤️): I agree! 👍

ParaskP7 commented on gradle/libs.versions.toml:
Minor (🔍): Some things I noticed during review, some related, some unrelated to the actual changes in this PR:

- Related: The `androidx-appcompat` version is pointing to `1.4.2` but the `login` library was pointing to `1.6.1`, we could consider updating it to point to a latest version instead of an older one. PS: I am sure that transitive updates already pointing to a much newer version, but... 🤷 
- Unrelated: The `androidx-activity` version is unused, we could consider removing it.
- Unrelated: The `android-billingclient-ktx` and `bumptech-glide-ksp` libraries are unused, we could consider removing them.

wzieba replied:
> Related: The androidx-appcompat version is pointing to 1.4.2 but the login library was pointing to 1.6.1, we could consider updating it to point to a latest version instead of an older one. PS: I am sure that transitive updates already pointing to a much newer version, but... 🤷

Sure! Good point, I took the version from FluxC but I forgot the higher could've been (and was) in Login.

ParaskP7 commented on gradle/libs.versions.toml:
```diff
@@ -77,19 +84,23 @@ lottie = '5.2.0'
-wordpress-fluxc = 'trunk-657a72d968f89dc2edd6a6e6e57f579643d1e213'
-wordpress-login = 'trunk-706a58b33ff8f1a2858ae64e51204426ec12fe7b'
 wordpress-libaddressinput = '0.0.2'
+wordpress-lint = "2.0.0"
```
Minor (🔍): The `wordpress-lint` version is pointing to `2.0.0` but the `login` library was pointing to `2.1.0`, we could consider updating it to point to a latest version instead of an older one.

ParaskP7 commented on gradle/libs.versions.toml:
```diff
@@ -77,19 +84,23 @@ lottie = '5.2.0'
 ksp = '2.0.21-1.0.27'
-mockito-inline = '4.6.1'
-mockito-kotlin = '4.0.0'
+mockito = '5.14.2'
```
Praise (❤️): Kudos for the upgrade! 🥇

ParaskP7 commented on settings.gradle:
```diff
@@ -77,6 +77,9 @@ include ':libs:commons'
 include ':libs:apifaker'
 include ':WooCommerce'
+include ':libs:fluxc', ':libs:fluxc-annotations', ':libs:fluxc-processor', ':libs:fluxc-plugin', ':libs:fluxc-tests'
+include ':libs:login'
```
Suggestion (💡): Maybe we could take this opportunity and group all `include` together, like:

```
include ':libs:commons'
include ':libs:cardreader'
include ':libs:apifaker'
include ':libs:fluxc', ':libs:fluxc-annotations', ':libs:fluxc-processor', ':libs:fluxc-plugin', ':libs:fluxc-tests'
include ':libs:login'
include ':WooCommerce'
include ':WooCommerce-Wear'
```

Or, further group them like you did with fluxc, on some kind of a logic, like `app` modules together, etc:

```
include ':libs:commons', ':libs:cardreader', ':libs:apifaker'
include ':libs:fluxc', ':libs:fluxc-annotations', ':libs:fluxc-processor', ':libs:fluxc-plugin', ':libs:fluxc-tests'
include ':libs:login'
include ':WooCommerce', ':WooCommerce-Wear'
```

ParaskP7 commented on libs/fluxc-tests/build.gradle:
Minor (🔍): We could consider getting rid of the `testInstrumentationRunner` from this module.

PS: Unrelated, but same applies for the `apifaker` and `libs:cardreader` modules.

ParaskP7 replied:
1. Minor (🔍): Consider removing the unnecessary `parcelize` and `kapt` plugins.
2. Minor (🔍): Consider removing the unnecessary `versionCode` and `versionName` configuration.
3. Minor (🔍): Consider removing the unnecessary `buildTypes { ... }` configuration.

ParaskP7 commented on libs/fluxc/build.gradle:
```diff
@@ -1,21 +1,22 @@
     namespace "org.wordpress.android.fluxc"
 
-    compileSdkVersion rootProject.compileSdkVersion
+    compileSdkVersion gradle.ext.compileSdkVersion
```
Minor (🔍): Consider replacing this with `compileSdk` and moving it within the `defaultConfig { ... }` next to other such configurations.

PS: Same applies for all such modules that are still using `compileSdkVersion` instead of `compileSdk`.

ParaskP7 commented on libs/fluxc/build.gradle:
Minor (🔍): Consider removing the `lint { ... }` related configuration from this module. It is related to `UnknownNullness` and pointing to a non existing `./config/lint/lint.xml` file.

PS: Same applies for all such modules that are still using this `lint { ... }` configuration, namely the `fluxc-plugin` and `fluxc-tests` modules.

ParaskP7 commented on libs/fluxc/build.gradle:
Suggestion (💡): Consider replacing the `gradle.properties-example` and `gradle.properties` related configuration with the `developer.properties-example` and `developer.properties`, just like you already did within the `login` module, following this similar change you did on [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/libs/fluxc/build.gradle#L29).

`developer.properties.example`
```
ENABLE_WPAPI=false
WP_ENABLE_DATABASE_DOWNGRADE=true
```

ParaskP7 commented on libs/fluxc/build.gradle:
```diff
@@ -63,82 +69,75 @@ android.buildTypes.all { buildType ->
+    implementation libs.apache.commons.text
+    api libs.androidx.paging.runtime
+    implementation libs.androidx.room.runtime
+    kapt libs.androidx.room.compiler
```
Suggestion (💡): Not sure if we want to do it now, but we could replace `kapt` with `ksp` for `room` (same for `dagger`) following this similar change you did on [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/libs/fluxc/build.gradle#L23), adding the `room` plugin and all.

PS: Same applies for the `fluxc-plugin` module.

ParaskP7 commented on libs/fluxc/build.gradle:
```diff
@@ -63,82 +69,75 @@ android.buildTypes.all { buildType ->
 
     // Encrypted Logging
     api "com.goterl:lazysodium-android:5.0.2@aar"
     api "net.java.dev.jna:jna:5.5.0@aar"
```
Minor (🔍): For these dependencies above you could consider moving them to `libs.versions.toml` following this similar change you did on [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/libs/fluxc/build.gradle#L111).

ParaskP7 commented on libs/fluxc-annotations/build.gradle:
```diff
@@ -1,26 +1,11 @@
+    targetCompatibility libs.versions.java.get()
 
     withSourcesJar()
     withJavadocJar()
```
Minor (🔍): Consider removing this configuration above.

PS: Same applies for the `fluxc-processor` module.

ParaskP7 commented on libs/fluxc-plugin/build.gradle:
Question (❓): I want to question this `testInstrumentationRunner` here and this one `MigrationTests` related test.

1. I am not sure it is being kept up-to-date, see the latest test being the `testMigration36to37` with `WC_DATABASE_VERSION` pointing to `38` already. 🤔
2. Maybe we should remove this UI/connected test and any `androidTest` configuration. And if not, maybe we should make this test part of the CI pipeline? 🤔

wzieba replied:
It's actually "in progress" ([internal link](https://appsinfrarequests.wordpress.com/2022/04/13/run-woocommerce-module-instrumentation-tests-on-ci-in-fluxc/)) 😄. I'd leave this as it is right now (it was like this in FluxC already) and addressed this after vendor is completed 👍

ParaskP7 commented on libs/fluxc-tests/src/test/java/org/wordpress/android/fluxc/comment/CommentSqlUtilsTest.kt:
```diff
@@ -239,9 +239,9 @@ class CommentSqlUtilsTest {
-        remoteComments.removeFirst()
+        remoteComments.removeAt(0)
+        remoteComments.removeAt(0)
+        remoteComments.removeAt(0)
```
Praise (❤️): Nicely done!

FYI: I've searched through the codebase and I found this last reference of [RollingLogEntries.add(...) -> removeFirst()](https://github.com/woocommerce/woocommerce-android/blob/cdcdbdca6d6863faa6c29679a1c0655f9e82a38c/WooCommerce/src/main/kotlin/com/woocommerce/android/util/RollingLogEntries.kt#L24). Maybe something to look into? 🤔

wzieba replied:
It's an interesting case, thanks for mentioning it! Actually, in this usage it's okay to use `removeFirst`. It's because `RollingLogEntries` inherits from `LinkedList`, which has its own implementation of `removeFirst` and doesn't have the same conflict as `List#removeFirst`.

I think I'd leave this as it is, especially that this class is covered well with unit tests, so we'll be notified if there will ever be a problem with it.

ParaskP7 replied:
Thanks for looking into that @wzieba ! 🙇

ParaskP7 commented on libs/login/build.gradle:
```diff
@@ -1,123 +1,90 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
 plugins {
```
Suggestion (💡): How about converting this to `alias` to keep the consistency between modules:

```
    alias(libs.plugins.android.library)
    alias(libs.plugins.kotlin.android)
    alias(libs.plugins.kotlin.kapt)
```

ParaskP7 commented on libs/login/build.gradle:
```diff
@@ -1,123 +1,90 @@
 plugins {
     id "com.android.library"
     id "org.jetbrains.kotlin.android"
     id "org.jetbrains.kotlin.kapt"
```
Suggestion (💡): Replacing `kapt` with `ksp` for this module is quite simple, maybe we should consider it.

PS: Then, we could also remove the `allWarningsAsErrors = false` related configuration from this module.

ParaskP7 reviewed (approved):
👋 @wzieba !

I have reviewed and (quickly) tested this PR as per the instructions, everything works as expected, another amazing job well done, kudos! 🌟 x 🌟 ^ 🌟 

PS: Thanks for adding the `script` you run for the extractions/filtering of `fluxc` and `login`.

-----

I have left a question (❓), a few suggestions (💡) and some minor (🔍) comments for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

-----

`EXTRAS`
1. About all the comments I added in my review, 18 in total, I know, but all I did is to gather the `TODOs` for you. I am sure that you already planned for most of those changes, but I didn't want to risk missing some of those, losing par with [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android).
2. I didn't do a thorough smoke test of the project, just a quick one. I am hoping that, after you apply any of the comment included in this review, someone from the WCAndroid team will jump on it with a thorough smoke test and give us a final 🟢 light for the merge.

ParaskP7 commented:
Minor (🔍): Btw @wzieba I forgot to mention in my review above that we would also want to also remove the [local-builds.gradle](https://github.com/woocommerce/woocommerce-android/blob/trunk/local-builds.gradle-example#L17-L18) configuration for both the `localFluxCPath` and `localLoginFlowPath`.

ParaskP7 commented:
> The current state of PR is based on code from following commits
> - FluxC: 657a72d968f89dc2edd6a6e6e57f579643d1e213
> - Login: 706a58b33ff8f1a2858ae64e51204426ec12fe7b

Another thing I forgot to mention @wzieba is that FluxC's trunk is already [3 commits](https://github.com/wordpress-mobile/WordPress-FluxC-Android/commits/trunk/) ahead:

![image](https://github.com/user-attachments/assets/8442e984-d116-4af6-8fff-adc2f55aa67b)

PS: Login's trunk is still on par.

wzieba committed:
Remove unused source publication declarations

wzieba committed:
Match androidx-appcompat version from Login instead of FluxC

wzieba committed:
Match wordpress-lint version from Login instead of FluxC

wzieba committed:
Remove not needed configuration from `fluxc-tests`

These were leftovers from when this module was example app

wzieba committed:
Group modules declarations

wzieba committed:
Login: adjust plugin configuration, replace kapt with ksp

wzieba committed:
Use `compileSdk` instead of deprecated `compileSdkVersion`

wzieba committed:
Remove not needed `lint` extension from fluxc modules

ParaskP7 commented:
Thanks for all the changes so far @wzieba , applying most of the suggestions, everything LGTM! 🥇

wzieba committed:
Use `ksp` when it's possible in `fluxc` module

Plus start using Room Gradle Plugin to declare schema directory

wzieba committed:
Move Encrypted Logs dependencies to version catalog

wzieba committed:
Remove include builds configuration

wzieba commented:
Thanks @ParaskP7 ! I've implemented now all improvements except one, I'll follow up in a next PR. Just a heads up: I'll be updating now FluxC branch and unfortunately it means it's possible that I'll have to rebase this PR (or cherry pick one by one commits here) and it will likely result with force push.

ParaskP7 commented:
Awesome stuff @wzieba and thanks for applying all these extra changes! 🥇 

After reviewing them carefully, I have the following for you:

1. Minor (🔍): About ad33a4cbfcfed327e6961e570ef2869a389ba5b9, consider adding `alias(libs.plugins.room) apply false` on the root level `build.gradle` file and its plugin configuration. 🫣
2. Minor (🔍): About de92f125350fc5b3a174291e076a373d081d08c5, I noticed that you copy-pasted the configuration from [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android), but in doing so, you are now using a higher `5.15.0@aar` version for `jna`, that is, instead of the `5.5.0@aar` version that was defined for this project. Maybe this is an improvement, but also, maybe we don't want to take any chances updating this library as part of this PR. 🤔 
3. Question (❓): About efc2ba9cc5cc463690cf5b93e1f36824ddc0fef2, and although I understand us removing the included build for the `fluxc` and `login` libraries, them being vendored and all, what I don't understand is why we are removing `mediatePicker` as well. Can you elaborate a bit on that? 🙏

wzieba committed:
Revert "Remove include builds configuration"

This reverts commit efc2ba9cc5cc463690cf5b93e1f36824ddc0fef2.

wzieba committed:
Remove only FluxC and Login libraries from include builds config

wzieba committed:
Provide Room Gradle Plugin to classpath of the root project

As it's suggested in the documentation https://developer.android.com/jetpack/androidx/releases/room#gradle-plugin

wzieba committed:
Revert JNA to 5.5.0

To not introduce any not required dependency changes in this PR

ParaskP7 commented:
FYI: Thanks for the recent changes just pushed @wzieba , did a quick review and everything LGTM! 💯 🙇 🚀

wzieba committed:
Measure FluxC and Login modules code coverage

wzieba commented:
TODO
- [ ] Remove FluxC tags
- [ ] Remove Login tags

wzieba committed:
Remove mentions of Mockito Core

It was removed on trunk, this is post-rebase fix

wzieba committed:
Revert Dagger downgrade

Another post-rebase fix

wzieba committed:
Migrate to Kotlin's `lowercase` and `uppercase` methods

This is now a compilation crashing error so the FluxC code has to be adjusted

wzieba commented:
@ParaskP7 since your last review I've committed following:
- 813143e
- a698386
- 4793192

And rebased this branch over new branches of FluxC and Login vendors to match the current versions of this libraries that Woo is using.

I think this PR is getting close to be merged! Do you have any last words? 😄

ParaskP7 commented:
> https://github.com/woocommerce/woocommerce-android/commit/813143e07f4dc591e2be9679b5dbf1350e6952e1

Thanks for the fix! 👍 

> a698386

👍 

> https://github.com/woocommerce/woocommerce-android/commit/4793192919457c903ff38f867f3372c4eaa6092f

I reviewed this commit and everything LGTM! 👍 

> And rebased this branch over new branches of FluxC and Login vendors to match the current versions of this libraries that Woo is using.

Awesome! 🚀 

> I think this PR is getting close to be merged! Do you have any last words? 😄

Let's do it @wzieba , we'll then have a whole week (4 days, ok, okay) to fix any problems that might arise till next release (`21.7 due by February 7`).

ParaskP7 commented:
Btw @wzieba , on the `fluxc` side, are we sure we brought in this [#3128](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3128/commits) change too? 🤔 

FYI: On the `login` side, everything looks okay.

wzieba commented:
> Btw @wzieba , on the fluxc side, are we sure we brought in this [#3128](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3128/commits) change too? 🤔

Good point! No, the PR is not part of the migration. And the reason is that this PR you linked is not part of 2.103.0 that WooCommerce Android is using right now.

I'd interpret the goal of this PR as strictly move a specific FluxC that WooCommerce is using at this very moment to not introduce yet another variable.

I think we'll need to backport this change after the merge. WDYT?

ParaskP7 commented:
> Good point! No, the PR is not part of the migration. And the reason is that this PR you linked is not part of 2.103.0 that WooCommerce Android is using right now.

👍 

> I'd interpret the goal of this PR as strictly move a specific FluxC that WooCommerce is using at this very moment to not introduce yet another variable.

👍 

> I think we'll need to backport this change after the merge. WDYT?

If we are fine backporting this change after the merge @wzieba (cherry-picking the change and all), then I am 👍 will it as well.

wzieba committed:
Merge branch 'trunk' into adjust_fluxc_login_vendor
